src/game/stats.c

```diff
@@ -0,0 +1,305 @@
+#include "game/stats.h"
+
+#include "game/draw.h"
+#include "game/gameflow.h"
+#include "game/input.h"
+#include "game/output.h"
+#include "game/savegame.h"
+#include "game/screen.h"
+#include "game/shell.h"
+#include "game/text.h"
+#include "global/vars.h"
+#include "log.h"
+
+#include <stdio.h>
+#include <string.h>
+
+#define PIERRE_ITEMS 3
+#define SKATEKID_ITEMS 1
+#define COWBOY_ITEMS 1
+#define BALDY_ITEMS 1
+
+static int32_t m_LevelPickups = 0;
+static int32_t m_LevelKillables = 0;
+static int32_t m_LevelSecrets = 0;
+static bool m_KillableItems[MAX_ITEMS] = { 0 };
+static bool m_IfKillable[O_NUMBER_OF] = { 0 };
+
+int16_t m_PickupObjs[] = { O_PICKUP_ITEM1,   O_PICKUP_ITEM2,  O_KEY_ITEM1,
+                           O_KEY_ITEM2,      O_KEY_ITEM3,     O_KEY_ITEM4,
+                           O_PUZZLE_ITEM1,   O_PUZZLE_ITEM2,  O_PUZZLE_ITEM3,
+                           O_PUZZLE_ITEM4,   O_GUN_ITEM,      O_SHOTGUN_ITEM,
+                           O_MAGNUM_ITEM,    O_UZI_ITEM,      O_GUN_AMMO_ITEM,
+                           O_SG_AMMO_ITEM,   O_MAG_AMMO_ITEM, O_UZI_AMMO_ITEM,
+                           O_EXPLOSIVE_ITEM, O_MEDI_ITEM,     O_BIGMEDI_ITEM,
+                           O_SCION_ITEM,     O_SCION_ITEM2,   O_LEADBAR_ITEM,
+                           NO_ITEM };
+
+// Pierre has special trigger check
+int16_t m_KillableObjs[] = {
+    O_WOLF,     O_BEAR,        O_BAT,      O_CROCODILE, O_ALLIGATOR,
+    O_LION,     O_LIONESS,     O_PUMA,     O_APE,       O_RAT,
+    O_VOLE,     O_DINOSAUR,    O_RAPTOR,   O_WARRIOR1,  O_WARRIOR2,
+    O_WARRIOR3, O_CENTAUR,     O_MUMMY,    O_ABORTION,  O_DINO_WARRIOR,
+    O_FISH,     O_LARSON,      O_SKATEKID, O_COWBOY,    O_BALDY,
+    O_NATLA,    O_SCION_ITEM3, O_STATUE,   O_PODS,      O_BIG_POD,
+    NO_ITEM
+};
+
+static void Stats_CheckTriggers(FLOOR_INFO **floor_array);
+
+static void Stats_CheckTriggers(FLOOR_INFO **floor_array)
+{
+    uint32_t secrets = 0;
+
+    for (int i = 0; i < g_RoomCount; i++) {
+        ROOM_INFO *r = &g_RoomInfo[i];
+        for (int x_floor = 0; x_floor < r->x_size; x_floor++) {
+            for (int y_floor = 0; y_floor < r->y_size; y_floor++) {
+
+                if (x_floor == 0 || x_floor == r->x_size - 1) {
```

Comment:
This is too nested; let's introduce a function for this.

Reply:
Just this part?
```
                if (x_floor == 0 || x_floor == r->x_size - 1) {
                    if (y_floor == 0 || y_floor == r->y_size - 1) {
                        continue;
                    }
                }
```

Reply:
No, that won't save us much; I meant the entire body of the third inner loop.